homeassistant/components/snmp/sensor.py

```diff
@@ -210,4 +210,4 @@ async def async_update(self):
             self.value = self._default_value
         else:
             for resrow in restable:
-                self.value = str(resrow[-1])
+                self.value = str(resrow[-1].prettyPrint())
```

Comment:
Why cast it to string?

and is this a breaking change?

Reply:
It's not a breaking change at all. It's just fixing the issue for non-number/string SNMP OIDs (string and numbers OIDs still return the same with of without `prettyPrint()`)

prettyPrint should return a string so it seems safe without it. Do you want me to remove it?

Reply:
I looked a bit into this, and from what I can tell pysnmp returns a [pysnmp.proto.rfc1902.IpAddress](https://github.com/etingof/pysnmp/blob/master/pysnmp/proto/rfc1902.py#L295) object in this case, which is inherited from `univ.OctetString` from `pyasn1`. Its `__str__()` implementation seems to implement a backward compatible behavior for Python2: https://github.com/etingof/pyasn1/blob/master/pyasn1/type/univ.py#L920.

I can confirm that so far, when using a OID which returned e.g. a IpAddress type, the sensor had a value which looks like the UTF-8 representation of that binary value, which is really nothing sensible. So I think this change can count as bug fix.

> prettyPrint should return a string so it seems safe without str. Do you want me to remove it?
Yeah I would suggest to remove it, `prettyPrint()` returns a string already.

Reply:
Thanks, done.